include/fix8/sessionwrapper.hpp

```diff
@@ -248,10 +248,16 @@ class ClientSession : public ClientSessionBase
 	/// Dtor.
 	virtual ~ClientSession ()
 	{
-		delete _persist;
+		session_ptr()->clear_connection( _cc );
 		delete _session;
-		delete _log;
+		_session = nullptr;
+		_persist->stop();
```

Comment:
Added comments to pull request.

________________________________
From: David Dight
Sent: Tuesday, March 10, 2015 8:06 PM
To: fix8/fix8
CC: "Stern, Andrew"
Subject: Re: [fix8] Fixed cleanup of sessionwrapper object. (#146)



In include/fix8/sessionwrapper.hpp<https://github.com/fix8/fix8/pull/146#discussion_r26179115>:

>               delete _session;
> -             delete _log;
> +             _session = nullptr;
> +             _persist->stop();


The ClientSession object is not designed to be reused. It should be instantiated for each Session, even on reconnect, unless you are using ReliableClientSession which does reuse Session but manages that itself.

—
Reply to this email directly or view it on GitHub<https://github.com/fix8/fix8/pull/146/files#r26179115>.
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-

This message is for the named person's use only. This communication is for informational purposes only and has been obtained from sources believed to be reliable, but it is not necessarily complete and its accuracy cannot be guaranteed. It is not intended as an offer or solicitation for the purchase or sale of any financial instrument or as an official confirmation of any transaction. Moreover, this material should not be construed to contain any recommendation regarding, or opinion concerning, any security. It may contain confidential, proprietary or legally privileged information. No confidentiality or privilege is waived or lost by any mistransmission. If you receive this message in error, please immediately delete it and all copies of it from your system, destroy any hard copies of it and notify the sender. You must not, directly or indirectly, use, disclose, distribute, print, or copy any part of this message if you are not the intended recipient. Any views expressed in this message are those of the individual sender, except where the message states otherwise and the sender is authorized to state them to be the views of any such entity.

Securities products and services provided to Canadian investors are offered by ITG Canada Corp. (member CIPF and IIROC - Investment Industry Regulatory Organization of Canada), an affiliate of Investment Technology Group, Inc.

Investment research products and services are produced and offered by ITG Investment Research, Inc. and not ITG Inc. (a FINRA member firm and SIPC member).

For CANADIANS Only: In accordance with the Canadian AntiSpam Legislation, ITG Canada has implied consent from you as a member of the investment community with whom we have already established a relationship through business discussions or dealings, or your email address was made available to ITG. However, if you are Canadian and wish to stop receiving ANY emails from ITG, please click on this button Unsubscribe <mailto:can-casl@itg.com?subject=Unsubscribe>. Please note that this does not apply if you are an existing client, and IIROC & OSC rules & regulations require us to continue to send you critical email communications

ITG Inc. and/or its affiliates reserves the right to monitor and archive all electronic communications through its network.

ITG Inc. Member FINRA, SIPC

-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-